Add and support metaclient.mangle_hostname configuration option

nphilipp opened the pull request:
This lets operators configure Jinja2 templates with which to
post-process hostnames before returning them to users of the legacy API.
The reason is that legacy Duffy only returned unqualified hostnames to
which some users simply appended ".ci.centos.org".

Fixes: #463

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp committed:
Add and support metaclient.mangle_hostname

This lets operators configure Jinja2 templates with which to
post-process hostnames before returning them to users of the legacy API.
The reason is that legacy Duffy only returned unqualified hostnames to
which some users simply appended ".ci.centos.org".

Fixes: #463

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp commented:
The latest change removes extensive logging configuration which referenced old locations of uvicorn logging formatters.

Background:  The `uvicorn.logging` module was renamed to `_logging` to not clash with the base module.

arrfab commented:
FWIW, even if all checks are failing for now, at the *functional* level I gave it a try and seems to be doing what's it's supposed to do. Let's verify later and then implement in a newer minor release to have it deployed on prod before migration in august

nphilipp committed:
Remove extensive logging configuration example

This is because uvicorn >= 0.18 renamed its logging module and marked it
as private, so we can't reference it from the configuration.
Furthermore, we didn’t really use it for anything ;).

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp commented:
My removing the `logging` section in the `app` and `metaclient` sections of the example configuration caused the tests to fail because some cases which formerly needed to remove parts of configuration in tests now need to add something to it. This is fixed now, I expect the tests in CI to succeed as well.
